crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -830,7 +850,11 @@ impl SlidingSync {
                                 .map(SlidingSyncView::request_generator)
                                 .collect();
                             *self.pos.lock_mut() = None;
-                            *self.extensions.lock_mut() = initial_extensions.clone();
+                            *self.extensions.lock().unwrap() = self.initial_extensions.clone();
```

Comment:
This is a noop now, `self.extensions` never changes and is always the same as `initial_extensions`.